utils/docker.go

```diff
@@ -0,0 +1,80 @@
+package utils
+
+import (
+	"crypto/rand"
+	"fmt"
+	"math/big"
+	"net"
+	"os"
+	"runtime"
+	"strings"
+
+	"github.com/ory/dockertest/docker"
+)
+
+// GetHostPort returns a resource's published port with an address.
+func GetHostPort(cont *docker.Container, portID string) string {
+	if cont == nil || cont.NetworkSettings == nil {
+		return ""
+	}
+
+	m, ok := cont.NetworkSettings.Ports[docker.Port(portID)]
+	if !ok || len(m) == 0 {
+		return ""
+	}
+
+	ip := m[0].HostIP
+	if ip == "0.0.0.0" {
+		ip = "localhost"
+	}
+	return net.JoinHostPort(ip, m[0].HostPort)
+}
+
+// RandLowerCaseLetterString returns a lowercase letter string of given length
+func RandLowerCaseLetterString(length int) string {
+	chars := []rune("abcdefghijklmnopqrstuvwxyz")
+	var b strings.Builder
+	for i := 0; i < length; i++ {
+		i, _ := rand.Int(rand.Reader, big.NewInt(int64(len(chars))))
```

Comment:
We probably don't -- I wouldn't say I've noticed consistent suffixes, so either it is getting seeded by a third party package, or there is enough nondeterminism in call order that they are different each time. 

If we change this part of the code, we can construct a rune or byte slice and call with `rand.Intn` instead of using a string builder and `math/big`.

Reply:
I agree. The `math/big` was a surprise to me.

Reply:
Feel free to clean this up if necessary. This has been carried forward from it's initial use in horcrux about 9 months ago.